### ISSUE TITLE
Key may not be empty

### ISSUE THREAD

**Issue by turowicz** [simpleidserver/EFCore.Cassandra#13]
I get the following exception when trying to create a schema.

![image](https://user-images.githubusercontent.com/1055391/97853227-20a3b980-1cf8-11eb-8dd6-1021b48271d4.png)

Context:

```    
    public class SqlContext : DbContext
    {
        private readonly IDbConnection _connection;

        public SqlContext()
        {
        }

        public SqlContext(IDbConnection connection)
        {
            _connection = connection;
        }

        public DbSet<UnitData> Header { get; set; }

        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        {
            if (!optionsBuilder.IsConfigured)
            {
                var connection = new CassandraConnectionStringBuilder();

                connection.Username = "cassandra";
                connection.Password = "cassandra";
                connection.ContactPoints = new string[] { "localhost" };

                optionsBuilder.UseCassandra(connection.ConnectionString, opt =>
                {
                    opt.MigrationsHistoryTable(HistoryRepository.DefaultTableName, "test");
                });
            }
        }

        protected override void OnModelCreating(ModelBuilder builder)
        {
            builder.UsePropertyAccessMode(PropertyAccessMode.Property);
            builder.ForCassandraAddKeyspace("test", new KeyspaceReplicationSimpleStrategyClass(1));

            builder.Entity<UnitData>().ToTable(nameof(UnitData), "test");
            builder.Entity<UnitData>().Ignore(table => table.ETag);
            builder.Entity<UnitData>().Ignore(table => table.Timestamp);
            builder.Entity<UnitData>().Ignore(table => table.Values);
            builder.Entity<UnitData>().ForCassandraSetClusterColumns(table => new { table.RowKey });
            builder.Entity<UnitData>().HasKey(table => new
            {
                table.PartitionKey,
                table.RowKey
            });
        }
    }

**Comment by turowicz** [comment]
cc @simpleidserver

**Comment by pluskal** [comment]
I have a feeling that this is a duplicate of #10.

**Comment by turowicz** [comment]
@pluskal I have seen that issue and have created the keyspace manually. I am trying to create tables automatically.

**Comment by simpleidserver** [comment]
We couldn't reproduce your issue. Can-you use the Nuget version "2.0.3" ?

**Comment by turowicz** [comment]
@simpleidserver I get the same error on `2.0.3`.

I have used the EF Migrations instead to sort this out. The bug is still there though.

**Comment by simpleidserver** [comment]
Sorry to answer you late, I was working on the *CaseManagement* Project. 
Can-you transfer your project ? I will  take a closer look :)

**Comment by Delta-38** [comment]
Hi, I've been seeing the same behaviour, calling EnsureCreated or EnsureCreatedAsync results in the mentioned exception. 
This can be verified even with the sample project when calling dbContext.Database.EnsureCreated().

**Comment by turowicz** [comment]
@simpleidserver the code pasted above is the project. The Program.cs only creates the context and calls EnsureCreatedAsync.

**Comment by simpleidserver** [comment]
The issue is fixed in the branch "release/2.0.4" :)

**Comment by Delta-38** [comment]
> The issue is fixed in the branch "release/2.0.4" :)

I confirm it fixed the issue on my copy of the sample and on a personal project as well.

**Comment by turowicz** [comment]
awesome!
